src/transforms/throttle.rs

```diff
@@ -0,0 +1,118 @@
+use crate::config::{DataType, TransformConfig, TransformContext, TransformDescription};
+use crate::event::Event;
+use crate::transforms::{TaskTransform, Transform};
+
+use async_stream::stream;
+use futures::{stream, Stream, StreamExt};
+use governor::*;
+use serde::{Deserialize, Serialize};
+use std::num::NonZeroU32;
+use std::pin::Pin;
+use std::time::Duration;
+
+#[derive(Deserialize, Default, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields, default)]
+pub struct ThrottleConfig {
+    events_per_second: u32,
+    bytes_per_second: u32,
+    key_field: Option<String>,
+    //behavior: Behavior,
+}
+
+//#[derive(Clone, Debug)]
+//enum Behavior {
+//    ShedLoad,
+//    BackPressure,
+//}
+
+inventory::submit! {
+    TransformDescription::new::<ThrottleConfig>("throttle")
+}
+
+impl_generate_config_from_default!(ThrottleConfig);
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "throttle")]
+impl TransformConfig for ThrottleConfig {
+    async fn build(&self, _context: &TransformContext) -> crate::Result<Transform> {
+        Throttle::new(self).map(Transform::task)
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Any
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Any
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "throttle"
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Throttle {
+    events_per_second: NonZeroU32,
+    bytes_per_second: NonZeroU32,
```

Comment:
I'm curious what others think about making this window configurable? Ex: per minute, etc.

Reply:
👍  `governor` has built-in for second, minute, hour - as well as providing custom timeframes but I found this the easiest to understand in terms of user configuration.

Reply:
Yeah, it is easier to understand, but I want to make sure we're able to meet requirements for users. These limits should align with whatever limits make the most sense for the use case:

* Teams might have daily quotas, which allows for more fluctuation.
* Services might have hourly quotas
* ...etc